keps/sig-apps/3329-retriable-and-non-retriable-failures/README.md

```diff
@@ -614,6 +635,46 @@ Go in to as much detail as necessary here.
 This might be a good place to talk about core concepts and how they relate.
 -->
 
+#### Failing delete after a condition is added
+
+Here we consider a scenario when a component fails (for example its container
+dies) between appending a pod condition and deleting the pod.
+
+In particular, in case of Scheduler it is possible that it decides to preempt
+a different pod the next time (or none). Then, this pod would be left with a
+condition that the pod was preempted, when it actually wasn't. This in turn
+could lead to inproper handling of the pod by the job controller.
+
+One possible solution to this issue would be to introduce a controller in
+kube-controller-manager that either completes a preemption (does a delete)
+or removes the condition (or set it's status to false) if the condition
+timestamp is old enough and there is no deletion timestamp.
+
+We are going to re-evaluate the need for solving this issue and the solution
+for Beta.
+
+#### Marking pods as Failed
+
+As indicated in by our experiments (see: [Disconnected node](#disconnected-node))
+a failed pod may get stuck in the `Running` phase when there is no kubelet
+working properly to change the phase. In particular, this happens
+in case of orphaned pods which are deleted, by garbage-collector. Due to this
+issue the logic of detecting pod failures in job controller is more complex than
+would be needed otherwise.
+
+Notably, this issue affects also an analogous scenario in which the taint-manager
+is disabled [Disconnected node when taint-manager is disabled](#disconnected-node-when-taint-manager-is-disabled).
+However, as disabling taint-manager is deprecated it is not a concern for this
```

Comment:
Does it mean that we should wait for Beta or at least GA promotion after that deprecation is completed?

Reply:
It would be nice to deprecate disabling taint manager before Beta and it's already started: https://github.com/kubernetes/kubernetes/pull/111411. However, I'm not sure if we need a formalized requirement for this feature, @alculquicondor ?

Reply:
If there are no concerns or oppositions, the deprecation would complete in 1.26.

Reply:
Call is out explicitly for GA promotion in that case.